### ISSUE TITLE
Conan config install failing randomly for parallel builds. 

### ISSUE THREAD

**Issue by mkjkec2005** [conan-io/conan#15018]
Hi, 

Conan config install is failing randomly for parallel builds. 

Command used: conan config install ${ARTIFACTORY_BASEURL}/artifactory/conanrepo/config.zip
----------------------------------------------------------------------------
[2023-10-24T23:42:43.945Z] Trying to download  https://artifactory-pd.com/artifactory/conanrepo/config.zip
[2023-10-24T23:42:43.945Z] ERROR: Failed conan config install: Error while installing config from https://artifactory-pd.com/artifactory/conanrepo/config.zip
[2023-10-24T23:42:43.945Z] [Errno 2] No such file or directory: '/home/bangbuild/.conan/tmp_config_install/config.zip'
script returned exit code 1

oR 

[2023-10-24T17:45:47.087Z] ERROR: Failed conan config install: [Errno 17] File exists: '/home/bangbuild/.conan/tmp_config_install'
script returned exit code 1

We are running builds in parallel inside docker containers. Conan cache is under /home/bangbuild (/home/bangbuild path in host is mounted inside the Docker container).

Regards,
Mohan 

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @mkjkec2005 

Thanks for your feedback.
This is expected, the Conan cache is not designed for concurrency. Every job should have its own cache, or run sequentially.

Conan 1.X had some limited support for a few operations, but ``conan config install`` was not one of them. Conan 2.0 doesn't have any concurrency support (you might find this explicitly stated in https://docs.conan.io/2/knowledge/guidelines.html), but it is planned to try to add them in the close future.

**Comment by mkjkec2005** [comment]
So, the only option is to export the CONAN_USER_HOME variable in our script/Jenkins pipelines. Is this understanding correct?
Kindly confirm.

**Comment by memsharded** [comment]
Yes, at this moment setting CONAN_USER_HOME to different folders (CONAN_HOME in Conan 2.0), is the recommended way for parallel builds.

**Comment by memsharded** [comment]
Closing in favor of https://github.com/conan-io/conan/issues/15840 where the Conan 2 cache concurrency future progress will be reported.
